### ISSUE TITLE
Roslyn .NET host takes up a lot of memory (likely background analysis scope related)

### ISSUE THREAD

**Issue by arkalyanms** [dotnet/vscode-csharp#6998]
Refer to https://github.com/microsoft/vscode-dotnettools/issues/518

**Comment by ShreyasJejurkar** [comment]
Workaround as of now => https://github.com/microsoft/vscode-dotnettools/issues/518#issuecomment-2042059436

**Comment by rjgotten** [comment]
> Workaround as of now => [microsoft/vscode-dotnettools#518 (comment)](https://github.com/microsoft/vscode-dotnettools/issues/518#issuecomment-2042059436)

That's not a workaround for the CPU & memory usage of the Roslyn background analysis though.
But a workaround for terminating .NET hosts that are idling and continue to take up RAM.

**Comment by ShreyasJejurkar** [comment]
> That's not a workaround for the CPU & memory usage of the Roslyn background analysis though.

Agreed, that depends upon options configured for background analysis in VSCode. 

But this workaround can be used for issues like this. 
https://github.com/microsoft/vscode-dotnettools/issues/518#issuecomment-1792534901

**Comment by rjgotten** [comment]
> > That's not a workaround for the CPU & memory usage of the Roslyn background analysis though.
> 
> Agreed, that depends upon options configured for background analysis in VSCode.
> 
> But this workaround can be used for issues like this. [microsoft/vscode-dotnettools#518 (comment)](https://github.com/microsoft/vscode-dotnettools/issues/518#issuecomment-1792534901)

No, it can't.
I reported that issue because the host in question continued to eat CPU and RAM well into 15 minutes after the system was left idle. The .NET host keeps being tapped for output somehow, so it never idles, so setting that environment variable would not let it shut down either. At worst, it would - and another process would start up immediately after not only giving you the RAM consumption penalty, but also double up on any process spawning and cold-startup costs.
